lib/utils.sh

```diff
@@ -23,17 +23,40 @@ asdf_repository_url() {
   echo "https://github.com/asdf-vm/asdf-plugins.git"
 }
 
+get_install_type(){
+  local install_type
+  install_type=$(get_asdf_config_value "install_type")
+
+  echo "$install_type"
+}
+
+asdf_data_dir(){
+  local data_dir
+
+  if [ "$(get_install_type)" = "user" ]; then
+    data_dir="${HOME}/.asdf"
+  else
+    data_dir="$(asdf_dir)"
+  fi
+
+  echo "$data_dir"
+}
```

Comment:
I'd rather see this function pull the directory name from an environment variable, and default to something reasonable if it's not set. We could look for `$ASDF_DATA_DIR` and default to `"$HOME/.asdf"` if it is not set. The install type logic may limit us in the future as users may want to customize the directory installations are placed in.

Reply:
Alternatively we could look in the config file for the path, and allow the user to set it there. However reading the config file would make this function slower, and it's used by the shims, so we want to keep it as fast as possible. I'm leaning towards not putting anything in the config file for this.